bindings/muen/time.c

```diff
@@ -33,5 +35,25 @@ uint64_t solo5_clock_monotonic(void)
 /* return wall time in nsecs */
 uint64_t solo5_clock_wall(void)
 {
-    return tscclock_monotonic() + tscclock_epochoffset();
+  const struct muen_resource_type *const
+    region = muen_get_resource("time_info", MUEN_RES_MEMORY);
+  struct time_info_type *time_info;
+
+  if (!region) {
+    log(WARN, "Unable to retrieve Muen time info region\n");
+    return -1;
+  }
+
+  time_info = (struct time_info_type *)region->data.mem.address;
+
+  uint64_t epoch = 0;
+  cc_barrier();
+
+  do
+  {
+    epoch = time_info->tsc_time_base * 1000;
+    cc_barrier();
+  } while (epoch == 0);
+
+  return epoch;
```

Comment:
Since "epoch" is always returned unaltered, without adding the current tscclock offset as before, it will return the same value for the lifetime of the Solo5 unikernel, i.e. time stands still. While the old implementation is a kludge and incorrect itself, it is still preferable, since at least `solo5_clock_wall` returns an increasing value.

Reply:
Yes, that's what I thought. Again, I don't know about Muen but is it exist a way to get the host's wall clock system via Muen (something else than the `tsc_time_base`)? We can wait more to have something which works perfectly :+1:. The idea is to delete the TSC-based clock and replace it by the right "hypercall".

To explain, we noticed a clock dritf on MirageOS and, for certain applications, it can be a real problem.